cfme/tests/infrastructure/test_infra_quota.py

```diff
@@ -238,3 +237,113 @@ def test_tenant_quota_max_cpu_check(
     # assert row.last_message.text == 'Request exceeds maximum allowed for the following: \
     # (cpu - Used: 526 plus requested: 8 exceeds quota: 3))'
     assert row.reason.text == "Quota Exceeded"
+
+
+@pytest.mark.tier(1)
+def test_tenant_quota_max_memory_check(
+        provisioner, prov_data, template_name, provider, request, vm_name, bug):
+    """ Test Tenant Quota-Max Memory by UI.
+
+    Prerequisities:
+        * A provider set up, supporting provisioning in CFME
+
+    Steps:
+        * Set the tenant quota for memory by UI emforcement
+        * Open the provisioning dialog.
+        * Apart from the usual provisioning settings, set memory greater then tenant quota memory.
+        * Submit the provisioning request and wait for it to finish.
+        * Visit the requests page. The last message should state quota validation message.
+
+    Metadata:
+        test_flag: provision
+    """
+    memory_data = {'memory_cb': True, 'memory': 2}
+    roottenant = Tenant.get_root_tenant()
+    roottenant.set_quota(**memory_data)
```

Comment:
Setting and un-setting the quota shouldn't be done within a test but from a yield_fixture. In addition, I don't see where we unset the quota, which could potentially cause weird issues with any provisioning tests ran after this one. I see that there is a yield fixture with cleanup created for the 'test_group_\*' tests in this module but not for the 'test_tenant_\*' ones which all seem to operate within the root tenant.

We definitely need to add cleanup to all of the 'test_tenant_*' tests.